### PR TITLE
Support KIND cluster with workload identity

### PR DESCRIPTION
**Pull request by matthchr**
Useful for testing AAD required features.

Also moves our default ci tests to use workload identity w/ a managed identity rather than aad pod identity (which is deprecated) with a service principal.

**If applicable**:
- [ ] this PR contains documentation
- [x] this PR contains tests

### CODE REVIEW

File: scripts/v2/create-kind-wi-storage.sh [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+
+print_usage() {
+  echo "Usage: create-kind-wi-storage.sh -d <DIRECTORY> -p <PREFIX>"
```
Comment (theunrepentantgeek) [inline scripts/v2/create-kind-wi-storage.sh:12]: What does `wi` mean?

File: scripts/v2/create-kind-wi-storage.sh [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+  --file "${DIR}/openid-configuration.json" \
+  --name .well-known/openid-configuration
+
+azwi jwks --public-keys "${DIR}/sa.pub" --output-file "${DIR}/jwks.json"
+
+az storage blob upload \
+  --account-name "${AZURE_STORAGE_ACCOUNT}" \
+  --container-name "${AZURE_STORAGE_CONTAINER}" \
+  --file "${DIR}/jwks.json" \
+  --name openid/v1/jwks
```
Comment (theunrepentantgeek) [inline scripts/v2/create-kind-wi-storage.sh:81]: Missing EoF.

File: scripts/v2/deploy-testing-secret.sh [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+metadata:
+  name: aso-controller-settings
+  namespace: azureserviceoperator-system
+stringData:
+  AZURE_SUBSCRIPTION_ID: "$AZURE_SUBSCRIPTION_ID"
+  AZURE_TENANT_ID: "$AZURE_TENANT_ID"
+  AZURE_CLIENT_ID: "$AZURE_MI_CLIENT_ID"
+  USE_WORKLOAD_IDENTITY_AUTH: "true"
+EOF
+fi
```
Comment (theunrepentantgeek) [inline scripts/v2/deploy-testing-secret.sh:56]: Missing EoF.

File: scripts/v2/make-mi-fic.sh [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+
+function retry_create_role_assignment() {
+  until create_role_assignment; do
+    sleep 5
+  done
+
+}
+
+print_usage() {
+  echo "Usage: make-mi-fic-kind-wi-storage.sh -g <RESOURCE_GROUP> -i <ISSUER> -d <DIR>"
```
Comment (theunrepentantgeek) [inline scripts/v2/make-mi-fic.sh:21]: What do `mi` and `wi` mean? Also, doesn't match filename.

File: scripts/v2/make-mi-fic.sh [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+}
+
+while getopts 'g:i:d:' flag; do
+  case "${flag}" in
+    g) RESOURCE_GROUP="${OPTARG}" ;;
+    i) ISSUER="${OPTARG}" ;;
+    d) DIR="${OPTARG}" ;;
```
Comment (theunrepentantgeek) [inline scripts/v2/make-mi-fic.sh:28]: Don't you need to predefine these variables (above the loop) in order to prevent inheriting random(ish) env vars?

**Review by theunrepentantgeek — APPROVED** [review]
To the extent that I understood it, looks good.

File: scripts/v2/create-kind-wi-storage.sh (hunk 1 above)
Comment (matthchr) [reply scripts/v2/create-kind-wi-storage.sh:12]: shorthand for workload identity

File: scripts/v2/make-mi-fic.sh (hunk 4 above)
Comment (matthchr) [reply scripts/v2/make-mi-fic.sh:21]: managed identity/workload identity

File: scripts/v2/make-mi-fic.sh (hunk 5 above)
Comment (matthchr) [reply scripts/v2/make-mi-fic.sh:28]: Probably yes. Will fix

**Pull request merged**